cluster/node.go

```diff
@@ -0,0 +1,72 @@
+package cluster
+
+import (
+	"sync"
+	"time"
+)
+
+type NodeState int
+
+const (
+	NodeReady       NodeState = 0
+	NodeNotReady    NodeState = 1
+	NodeUnreachable NodeState = 2
+)
+
+var NodeStateText = map[NodeState]string{
+	NodeReady:       "Ready",
+	NodeNotReady:    "NotReady",
+	NodeUnreachable: "Unreachable",
+}
+
+type Node struct {
+	Name    string    `json:"name"`
+	Version string    `json:"version"`
+	Primary bool      `json:"primary"`
+	State   NodeState `json:"state"`
+	Started time.Time `json:"started"`
+	sync.RWMutex
```

Comment:
also all these members should probably be un-exported, since we want to make sure they are safely accessed through the methods, and not directly